File: src/stores/mail.ts
import { create } from "zustand";

export interface Email {
  id: string;
  from: string;
  to: string[];
  subject: string;
  htmlBody: string | null;
  textBody: string | null;
  timestamp: string;
  read: boolean;
  appName: string | null;
}

interface MailState {
  emails: Email[];
  setEmails: (emails: Email[]) => void;
  addEmail: (email: Email) => void;
  markRead: (id: string) => void;
  deleteEmail: (id: string) => void;
  clearAll: () => void;
}

export const useMailStore = create<MailState>((set) => ({
  emails: [],
  setEmails: (emails) => set({ emails }),
  addEmail: (email) => set((state) => ({ emails: [email, ...state.emails] })),
  markRead: (id) =>
    set((state) => ({
      emails: state.emails.map((e) =>
        e.id === id ? { ...e, read: true } : e
      ),
    })),
  deleteEmail: (id) =>
    set((state) => ({ emails: state.emails.filter((e) => e.id !== id) })),
  clearAll: () => set({ emails: [] }),
}));

File: src-tauri/src/commands/mail.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct EmailMessage {
    pub id: String,
    pub from: String,
    pub to: Vec<String>,
    pub subject: String,
    pub html_body: Option<String>,
    pub text_body: Option<String>,
    pub timestamp: String,
    pub read: bool,
    pub app_name: Option<String>,
}

#[tauri::command]
pub async fn get_emails() -> Result<Vec<EmailMessage>, String> {
    // TODO: Read from mail storage
    Ok(Vec::new())
}

#[tauri::command]
pub async fn delete_email(email_id: String) -> Result<(), String> {
    tracing::info!("Deleting email {}", email_id);
    // TODO: Delete email file
    Ok(())
}

#[tauri::command]
pub async fn clear_all_emails() -> Result<(), String> {
    tracing::info!("Clearing all emails");
    // TODO: Clear mail directory
    Ok(())
}
